apps/myaccount/src/pages/personal-info.tsx

```diff
@@ -83,10 +84,15 @@ const PersonalInfoPage:  FunctionComponent<PersonalInfoPagePropsInterface> = (
         const localCredentialExist = profileDetails?.profileInfo?.[SCIMConfigs.scim.customEnterpriseSchema]?.
             [ProfileConstants?.SCIM2_SCHEMA_DICTIONARY.get("LOCAL_CREDENTIAL_EXISTS")];
 
+        // Requires to validate if user is logged in from different IDP other than the source IDP
+        setUserSourceIdp(profileDetails?.profileInfo?.[SCIMConfigs.scim.customEnterpriseSchema]?.
+            [ProfileConstants?.SCIM2_SCHEMA_DICTIONARY.get("IDP_TYPE")]);
+
         if (localCredentialExist && localCredentialExist == "false") {
-            setIsNonLocalCredentialUser(true);
+            setIsNonLocalCredentialUser(false);
```

Comment:
what's the purpose of this change and is it correct?

Reply:
The line `84` will return `false` when there are no local credentials exist. Hence the condition on line `91` would become true. Which will make the variable `true` yet the place where the variable is checked ( line `262` : `federated-associations.tsx` ) checks for the negation of the variable. Which leads to the bug we come across. Hence this change will fix the flow.